src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -297,6 +310,43 @@ private void removeFeedbacksForAfterDueDateTests(Result result, Set<ProgrammingE
             result.setHasFeedback(false);
     }
 
+    /**
+     * Sets the category for each feedback and removes feedback with no or an inactive category
+     * @param result of the build run
+     * @param staticCodeAnalysisFeedback List of feedback objects
+     * @param programmingExercise The current exercise
+     * @return The filtered list of feedback objects
+     */
+    private List<Feedback> removeInvisibleScaFeedback(Result result, List<Feedback> staticCodeAnalysisFeedback, ProgrammingExercise programmingExercise) {
+        var categories = staticCodeAnalysisService.getCategoriesWithMappingForExercise(programmingExercise);
+
+        return staticCodeAnalysisFeedback.stream().filter(feedback -> {
```

Comment:
please add more inline comments

Reply:
It's unclear why you need an ObjectMapper here

Reply:
Added comments and improved the code readability.